internal/mode/static/nginx/runtime/manager.go

```diff
@@ -29,41 +32,48 @@ type (
 // Manager manages the runtime of NGINX.
 type Manager interface {
 	// Reload reloads NGINX configuration. It is a blocking operation.
-	Reload(ctx context.Context) error
+	Reload(ctx context.Context, configVersion int) error
 }
 
 // ManagerImpl implements Manager.
-type ManagerImpl struct{}
+type ManagerImpl struct {
+	verifyClient *verifyClient
+}
 
 // NewManagerImpl creates a new ManagerImpl.
 func NewManagerImpl() *ManagerImpl {
-	return &ManagerImpl{}
+	return &ManagerImpl{
+		verifyClient: newVerifyClient(nginxReloadTimeout),
+	}
 }
 
-func (m *ManagerImpl) Reload(ctx context.Context) error {
+func (m *ManagerImpl) Reload(ctx context.Context, configVersion int) error {
 	// We find the main NGINX PID on every reload because it will change if the NGINX container is restarted.
 	pid, err := findMainProcess(ctx, os.Stat, os.ReadFile, pidFileTimeout)
 	if err != nil {
 		return fmt.Errorf("failed to find NGINX main process: %w", err)
 	}
 
+	childProcFile := fmt.Sprintf("/proc/%[1]v/task/%[1]v/children", pid)
+	previousChildProcesses, err := os.ReadFile(childProcFile)
+	if err != nil {
+		return err
+	}
+
 	// send HUP signal to the NGINX main process reload configuration
 	// See https://nginx.org/en/docs/control.html
 	if err := syscall.Kill(pid, syscall.SIGHUP); err != nil {
```

Comment:
Part of the AC is to retry reloads on failure. Can that be done as a part of this PR or should we create a separate issue for it?

Reply:
Discussed offline, to recap: Most of the functionality we would like to achieve from retrying is covered. We wait for new processes to spin up, and then wait for the reload to succeed. If there is an error calling the signal or no new worker processes are created due to bad config, we likely don't want to retry.